### ISSUE TITLE
[COGNITION]: GIBCT School Locations - Tables SHOULD be responsive when the new component is ready

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#11055]
## [508-defect-3](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-3)

**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

The GIBCT school locations table is using a separate markup strategy for mobile width, that removes the semantic meaning of the `<table>` at mobile screen sizes. This was done out of necessity because there wasn't a good responsive table solution at the time. Work is now underway to build a [design system component for responsive tables](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/203) and this should be implemented when it's ready. Screenshot attached below.

## Point of Contact

**VFS Point of Contact:** _Trevor_

## Related Issues
* https://github.com/department-of-veterans-affairs/va.gov-team/issues/9953
* https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/203

## Acceptance Criteria

- [ ] The new responsive table is implemented when ready
- [ ] No new axe errors or visual regressions
- [ ] Existing functionality for setting focus on the correct table cell is maintained

## Environment

* https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/203

## Screenshots or Trace Logs

![Screen Shot 2020-07-08 at 4 34 37 PM](https://user-images.githubusercontent.com/934879/86972735-05a1ae00-c139-11ea-9865-7cf6eb530dbf.png)

**Comment by 1Copenut** [comment]
@desireeturner && @hughes-dustin I talked with the Design System Council on Friday about this item. We won't have a responsive component ready for a bit, but there is a team working on a responsive table that you can leverage. The eBenefits team has created a version on their Payments History view: https://staging.va.gov/va-payment-history/payments/#content

I'd like to use this code pattern and the CSS they're using to rearrange the table, instead of swapping the table for a list wholesale. The only catch, the table markup the eBenefits team has written needs `role` attributes applied to each element for the mobile version to work. When tables are assigned `display: block;` for elements, the table semantics are removed for screen readers. I've mocked up a simple pattern using the eBenefits team's code, with the roles added:

```html
<table class="responsive" role="table">
  <thead role="rowgroup">
    <tr role="row">
      <th class="vads-u-border-top--0..." role="columnheader">
      	Date
      </th>
      ...
    </tr>
  </thead>
  <tbody role="rowgroup">
    <tr class="vads-u-border-top--0..." role="row">
      <td
        class="vads-u-border-top--0..."
        data-index="0"
        data-label="Date:"
        role="cell"
      >June 4, 2020</td>
    </tr>
  </tbody>
</table>
```

LMK if you'd like to discuss further or have any questions. /cc @jenstrickland for visibility. Jennifer works closely with this team and has a lot of experience with responsive tables.

**Comment by hughes-dustin** [comment]
Developer notes: 

Copied over `src/applications/disability-benefits/view-payments/components/responsive-table/ResponsiveTable.jsx` to `src/applications/gi/components/ResponsiveTable.jsx`and still refactoring the locations table to use it. 

local branch is bah-11055.

**Comment by 1Copenut** [comment]
@hughes-dustin LMK when your branch is in the remote repo and I can take a look.

**Comment by elder-joseph** [comment]
Associated Pull Request

**Comment by sonntag-adam** [comment]
@1Copenut this is now in staging.

**Comment by 1Copenut** [comment]
@sonntag-adam This is a good start. The table should reflow to a single column like the screenshot below. It also must contain the roles I added in the HTML snippet above to retain table semantics. LMK if you want to discuss further or have questions.

---
![Screen Shot 2020-09-16 at 12.17.38 PM.png](https://images.zenhubusercontent.com/5ac217b74b5806bc2bcd3fc8/33a40e78-abad-4c4a-a08c-23b6afb750a3)

**Comment by zurbergram** [comment]
Associated defect https://github.com/department-of-veterans-affairs/va.gov-team/issues/13653

**Comment by zurbergram** [comment]
eslint errors for certain role attributes

![Screen Shot 2020-09-17 at 11 31 43 AM](https://user-images.githubusercontent.com/1094999/93492935-5e2fbd00-f8d9-11ea-89aa-000978b0608f.png)
![Screen Shot 2020-09-17 at 11 54 45 AM](https://user-images.githubusercontent.com/1094999/93495856-9684ca80-f8dc-11ea-8d29-3844d1e21ea2.png)

**Comment by 1Copenut** [comment]
@zurbergram We'll have to suppress the errors. We need those role attributes to maintain table semantics when we switch to `display: grid` at mobile sizes.

https://adrianroselli.com/2018/02/tables-css-display-properties-and-aria.html

**Comment by zurbergram** [comment]
https://github.com/department-of-veterans-affairs/vets-website/pull/14288#issuecomment-696408767

**Comment by 1Copenut** [comment]
@zurbergram I'd like to take one more pass at this, and add the `scope` attributes to TH elements. I added a code snippet and explanation in [my comment to PR 14288](https://github.com/department-of-veterans-affairs/vets-website/pull/14288#issuecomment-696408767). If that needs to be a separate ticket, I'll test the code as is.

Additional resources on the `scope` attribute:
* [Deque: Scope attribute should be used correctly on tables](https://dequeuniversity.com/rules/axe/3.5/scope-attr-valid)
* [MDN: Tables for visually impaired users](https://developer.mozilla.org/en-US/docs/Learn/HTML/Tables/Advanced#Tables_for_visually_impaired_users)

**Comment by zurbergram** [comment]
@1Copenut https://github.com/department-of-veterans-affairs/vets-website/pull/14338 for

> @zurbergram I'd like to take one more pass at this, and add the `scope` attributes to TH elements. I added a code snippet and explanation in [my comment to PR 14288](https://github.com/department-of-veterans-affairs/vets-website/pull/14288#issuecomment-696408767). If that needs to be a separate ticket, I'll test the code as is.
> 
> Additional resources on the `scope` attribute:
> 
> * [Deque: Scope attribute should be used correctly on tables](https://dequeuniversity.com/rules/axe/3.5/scope-attr-valid)
> * [MDN: Tables for visually impaired users](https://developer.mozilla.org/en-US/docs/Learn/HTML/Tables/Advanced#Tables_for_visually_impaired_users)

**Comment by delli-gatti-michael** [comment]
local testing complete

**Comment by delli-gatti-michael** [comment]
@1Copenut This is on staging and ready for your 👀

**Comment by 1Copenut** [comment]
@delli-gatti-michael I was able to give this a look yesterday morning on Staging. Everything worked as expected, so we can close this issue.

* https://github.com/department-of-veterans-affairs/vets-website/pull/14338#pullrequestreview-494736990
* https://github.com/department-of-veterans-affairs/vets-website/pull/14338#pullrequestreview-494966265
